### PR TITLE
[Mellanox] [201911] Support shared headroom pool

### PR DESCRIPTION
**Pull request by stephenxs**
**- Why I did it**
Support shared headroom pool on Mellanox platforms on the 201911 branch.

**- How I did it**
Update default buffer configurations and pg_profile_lookup

**- How to verify it**
Run QoS test, regression test

**- Which release branch to backport (provide reason below if selected)**

- [ ] 201811
- [ ] 201911
- [ ] 202006

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

**Comment by liat-grozovik** [comment]
retest broadcom please

**Comment by neethajohn** [comment]
@baiwei0427, please review the values used

**Comment by liat-grozovik** [comment]
@baiwei0427 could you please review and approve?

**Comment by baiwei0427** [comment]
> @baiwei0427 could you please review and approve?

Will review it tomorrow

File: device/mellanox/x86_64-mlnx_msn2700-r0/Mellanox-SN2700-D48C8/pg_profile_lookup.ini [hunk 1]
```diff
@@ -1,17 +1,17 @@
...
-  25000   40m 44032  19456  24576          0
-  40000   40m 45056  19456  25600          0
-  50000   40m 45056  19456  25600          0
- 100000   40m 49152  19456  29696          0
-  10000  300m 47104  19456  27648          0
-  25000  300m 56320  19456  36864          0
-  40000  300m 64512  19456  45056          0
-  50000  300m 69632  19456  50176          0
- 100000  300m 98304  19456  78848          0
+  10000    5m 19456  19456  22528          0
```
Comment (baiwei0427) [inline device/mellanox/x86_64-mlnx_msn2700-r0/Mellanox-SN2700-D48C8/pg_profile_lookup.ini:3]: We can use an unified Xoff value calculated based on the worst case, e.g., 100G and 2KM cables.

File: device/mellanox/x86_64-mlnx_msn2700-r0/Mellanox-SN2700-D48C8/buffers_defaults_t0.j2 [hunk 2]
```diff
@@ -1,7 +1,8 @@
 {% set default_cable = '5m' %}
-{% set ingress_lossless_pool_size =  '6687744' %}
+{% set ingress_lossless_pool_size =  '7719936' %}
+{% set ingress_lossless_pool_xoff =  '1032192' %}
 {% set egress_lossless_pool_size =  '13945824' %}
-{% set egress_lossy_pool_size =  '6687744' %}
+{% set egress_lossy_pool_size =  '7719936' %}
```
Comment (baiwei0427) [inline device/mellanox/x86_64-mlnx_msn2700-r0/Mellanox-SN2700-D48C8/buffers_defaults_t0.j2:5]: What assumptions do you hold when calculating buffer values for different SKUs?

**Review by baiwei0427 — CHANGES_REQUESTED** [review]

File: device/mellanox/x86_64-mlnx_msn2700-r0/Mellanox-SN2700-D48C8/buffers_defaults_t0.j2 (hunk 2 above)
Comment (stephenxs) [reply device/mellanox/x86_64-mlnx_msn2700-r0/Mellanox-SN2700-D48C8/buffers_defaults_t0.j2:5]: Hi, Wei,
FYI.
For 3800 platform:
- C64: 32 100G down links and 32 100G up links.
- D112C8: 112 50G down links and 8 100G up links.
- D24C52: 24 50G down links, 20 100G down links, and 32 100G up links.
- D28C50: 28 50G down links, 18 100G down links, and 32 100G up links.

For 2700 platform:
- Mellanox-SN2700-C28D8 t0, assume 48 * 50G/5m + 8 * 100G/40m ports
- Mellanox-SN2700 (C32)
  - t0: 16 * 100G/5m + 16 * 100G/40m
  - t1: 16 * 100G/40m + 16 * 100G/300m

**Review by baiwei0427 — APPROVED** [review]

**Review by neethajohn — APPROVED** [review]

**Review by liat-grozovik — APPROVED** [review]

**Pull request merged**